.github/workflows/dockerimage.yml

```diff
@@ -0,0 +1,47 @@
+name: Build Docker Image
+
+on:
+  push:
+    branches:
+      - master
+  release:
+    types: [published]
+  pull_request:
+    branches:
+      - master
+
+env:
+  DOCKER_BASE_NAME: docker.pkg.github.com/${{ github.repository }}/optuna
+  DOCKER_HUB_BASE_NAME: optuna/optuna
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python_version: ['3.5', '3.6', '3.7', '3.8']
+        build_type: ['', 'dev']  # empty string represents the normal optuna environment.
```

Comment:
How about rephrasing the comment?
At first, I didn't understand what 'normal' meant. 
The following is just an example:
 `Install Optuna with minimum requirements if empty string is specified.`

Reply:
Absolutely right.

Alternatively we could say `"dev" installs all the dependencies including pytest`.